### DISCUSSION TITLE
Should "test" emails work?

### DISCUSSION THREAD

**Discussion by richcorbs** [pockethost/pockethost#154] [Q&A] [answered]
I tried to send a test email from a pockethost.io instance and it gave me an error about not finding sendmail.

**Comment by osmely** [comment]
Hi @richcorbs .. I just tested my instances and it works for me :/

**Reply by richcorbs** [reply]
Thanks @osmely.

Did you have to configure anything after spinning up the instance?

Just to confirm this is at Settings > Mail Settings > Send test email in the UI.

**Reply by osmely** [reply]
yes, I use smtp mail server.

**Reply by osmely** [reply]
ohh, you are using sendmail.. sorry :)

**Reply by richcorbs** [reply]
I see this on the home page and thought sending emails should just work without extra configuration:

> Email and oAuth authentication options work out of the box. Send transactional email to your users from our verified domain and your custom address

Seems like I've misunderstood?

**Reply by osmely** [reply]
mmm I never tried sendmail, my bad... we need @benallfree :)

**Reply by benallfree** [reply]
> I see this on the home page and thought sending emails should just work without extra configuration:
> 
> > Email and oAuth authentication options work out of the box. Send transactional email to your users from our verified domain and your custom address
> 
> Seems like I've misunderstood?

Ah, that’s my fault. Marketing for 1.0 while still in anlpha.

**Comment by benallfree** [comment, marked as answer]
Hey everyone,

#24 may cover what you’re seeking. For now, I recommend setting up Amazon SES and using their outgoing smtp. 

Eventually, I want to make it so instances can send from <myinstance>@pockethost.io as long as it is to a registered user for that instance. I may roll that out early and see if people abuse it, and if they do, I’ll have to add a proxy to block abuse.

**Reply by ausminternet** [reply]
Any updates on this?

**Reply by benallfree** [reply]
Still on the todo list but getting closer. Want to work on it with me?

**Reply by ausminternet** [reply]
As a happy webdeveloper who's using pocketbase with the js-sdk and with absolutely zero knowladge about go: I'm probably not the right one :D

**Reply by benallfree** [reply]
PocketHost is built in nodejs ;)

**Reply by ausminternet** [reply]
Ha! Then I can at least have a look at what it would be about - no promises ;)

**Reply by benallfree** [reply]
See if you can find a nodejs-based SMPT proxy and test it out with a local copy of PocketBase. So in other words, PocketBase would be configured with the SMTP proxy credentials and then the SMPT proxy would be configured with your Amazon SES (or other email provider) credentials. 

If that works, see if you can do things like:
* enforce a rate limit
* add a footer to the message
* use an async call to validate the login and password - this can be a placeholder for now but it will eventually call into the pockethost database

**Reply by ausminternet** [reply]
"been there, done that" - what have you done so far, which proxies have you tried?

**Reply by benallfree** [reply]
I have not done anything or tested anything. I am aware of a few npm packages that claim to be SMTP proxies, but I have not even tried them.

smtp-server and nodemailer look like possibilities: https://npmtrends.com/nodemailer-vs-smtp-server
